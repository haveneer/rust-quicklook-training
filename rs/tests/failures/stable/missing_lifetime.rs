#![allow(unused_variables)]

fn f(x: &str, y: &str) -> &str {
    todo!()
}

fn main() {}
