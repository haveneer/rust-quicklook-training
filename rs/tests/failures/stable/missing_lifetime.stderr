error[E0106]: missing lifetime specifier
 --> tests/failures/stable/missing_lifetime.rs:3:27
  |
3 | fn f(x: &str, y: &str) -> &str {
  |         ----     ----     ^ expected named lifetime parameter
  |
  = help: this function's return type contains a borrowed value, but the signature does not say whether it is borrowed from `x` or `y`
help: consider introducing a named lifetime parameter
  |
3 | fn f<'a>(x: &'a str, y: &'a str) -> &'a str {
  |     ++++     ++          ++          ++
